homeassistant/components/envisalink/__init__.py

```diff
@@ -89,6 +91,10 @@
                     CONF_ZONEDUMP_INTERVAL, default=DEFAULT_ZONEDUMP_INTERVAL
                 ): vol.Coerce(int),
                 vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): vol.Coerce(int),
+                vol.Optional(
+                    CONF_CREATE_ZONE_BYPASS_SWITCHES,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
I thought there was still work needed upstream, but it seems the library has merged your PR and that was only to add an option.

Isn't possible to solve the problem without an option? Can we detect what the situation is and adjust the behavior accordingly?

Reply:
Oh..  and yes, the initial set of changes did not include the upstream package version as I wasn't sure when the package author was going to be able to push a new version and didn't want to leave the affected users here hanging for too long.  He was able to publish the package last night so I updated this PR to remove the hack and use the fix from the upstream version.

Reply:
Ok, then I suggest we disable the feature for now. It was only introduced in the last release so impact should not be huge. It's more important that existing features work.

Reply:
Okay, that seems a little unfortunate given we can have both working with this PR and I'll follow up with a separate PR to refactor the config flow but policies are policies I guess.

Is there an official technic required for disabling features?  Does all the related code need to be removed or is it acceptable to just prevent the functionality from being made available?

Reply:
If you plan to continue working on the integration I think it's ok to just disable the feature now. I think this means don't load the switch platform and remove the callback registration on the library panel instance.